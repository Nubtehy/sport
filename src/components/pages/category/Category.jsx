import React from 'react';
import PropTypes from 'prop-types';
import { Master } from 'components/composables';

import { Product } from 'components/widgets';
import { ProductWrapper } from './Category.styled';

const displayName = 'Category';

const propTypes = {

};

const defaultProps = {

};

function Category({
  myProducts,
  handleAddToCart,
  total
}) {
  const totalItems = total ? total.length : 0;

  return (
    <Master title="Products" total={ totalItems }>
      <ProductWrapper>

        {myProducts.map(product => (
          <Product key={product.id} {...product} AddToCart={handleAddToCart}/>
        ))}
      </ProductWrapper>
    </Master>
  );
}

Category.displayName = displayName;
Category.propTypes = propTypes;
Category.defaultProps = defaultProps;

export default Category;
