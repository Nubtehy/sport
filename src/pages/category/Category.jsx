import React from 'react';
import PropTypes from 'prop-types';
import { Master } from 'composables';

const displayName = 'Cart';

const propTypes = {

};

const defaultProps = {

};

function Category() {
  return (
    <Master>
      <h1>Category</h1>
    </Master>
  );
}

Category.displayName = displayName;
Category.propTypes = propTypes;
Category.defaultProps = defaultProps;

export default Category;
