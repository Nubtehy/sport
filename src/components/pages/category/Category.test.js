import React from 'react';
import { shallow } from 'enzyme';

import Category from './Category';

const testProps = {
  myProducts: [],
  handleAddToCart: jest.fn(),
  total: [],
};

describe('Given a Category component', () => {
  describe('when the component is rendered', () => {
    let component;

    beforeAll(() => {
      component = shallow(<Category {...testProps} />);
    });

    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
