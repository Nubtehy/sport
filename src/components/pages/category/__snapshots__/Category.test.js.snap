// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Category component when the component is rendered should match the snapshot 1`] = `
<Master
  title="Products"
  total={0}
>
  <Categorystyled__ProductWrapper />
</Master>
`;
